### PR TITLE
add :okhttp and :test-okhttp modules

### PR DESCRIPTION
**Pull request by dsvoronin**

### CODE REVIEW

File: okhttp/src/main/kotlin/com/avito/http/FallbackInterceptor.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            chain.doFallback(request)
+        } else {
+            response ?: throw exception!!
+        }
+    }
+
+    private fun isFallbackNeeded(response: Response?): Boolean {
+        return when (response) {
+            null -> true
+            else -> when {
```
Comment (eugene-krivobokov) [inline okhttp/src/main/kotlin/com/avito/http/FallbackInterceptor.kt]: Nested "when" looks overcomplicated for reading.
I'd rather split this check and return earlier.

File: okhttp/src/main/kotlin/com/avito/http/RequestBodies.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    return if (isPlaintext(buffer)) {
+        buffer.readString(charset!!)
+    } else {
+        null
+    }
+}
+
+/**
+ * Returns true if the body in question probably contains human readable text. Uses a small sample
```
Comment (eugene-krivobokov) [inline okhttp/src/main/kotlin/com/avito/http/RequestBodies.kt]: "in question" ?

**Review by eugene-krivobokov — APPROVED** [review]

File: okhttp/src/main/kotlin/com/avito/http/RequestBodies.kt (hunk 2 above)
Comment (dsvoronin) [reply okhttp/src/main/kotlin/com/avito/http/RequestBodies.kt]: no idea, fixed

File: okhttp/src/main/kotlin/com/avito/http/FallbackInterceptor.kt (hunk 1 above)
Comment (dsvoronin) [reply okhttp/src/main/kotlin/com/avito/http/FallbackInterceptor.kt]:
```
return when {
            response == null -> true
            response.isSuccessful -> false
            else -> response.code() in doFallbackOnTheseCodes
        }
```
idea's variant, looks readable

**Pull request merged**